libpod/options.go

```diff
@@ -550,6 +550,18 @@ func WithSystemd() CtrCreateOption {
 	}
 }
 
+// WithSystemd turns on systemd mode in the container
+func WithSdNotifyMode(mode string) CtrCreateOption {
+	return func(ctr *Container) error {
```

Comment:
I think we should validate `mode` here and make sure that the value is supported.

Reply:
Agree

Reply:
@vrothberg @mheon 
Validation already happened in pkg/specgen/container_validate.go

I used the systemd option as a model - that seemed to be where the validation belonged. (i.e. see WithSystemd() directly above this hunk)

Reply:
I can add it again if you want.